colossalai/inference/core/engine.py

```diff
@@ -667,6 +667,10 @@ def add_request(
             elif max_length is not None:
                 max_new_tokens = max_length - len(prompts_token_ids[i])
 
+            assert (
```

Comment:
Consider when enabling streaming-llm, how would a user set up ` max_output_length`, `max_new_tokens`, and `max_length`. Saying that (just a condition) a user set `max_output_length` as 1024 for inference engine init, use streaming-llm, and set `max_length` as 4096 for generation, would the assertion block the user from using?

Reply:
I think this situation should be avoided. When we do inference in the usual way, if max_output_length is less than max_new_tokens, it will cause the kvcache manager to report an error.

Reply:
Hence, conditions between using streaming-llm and those not using streaming-llm should be handled specifically.

Reply:
Done